File: scratch/extract_cvs.py
import sys
import os
try:
    from PyPDF2 import PdfReader
except ImportError:
    import subprocess
    subprocess.check_call([sys.executable, "-m", "pip", "install", "PyPDF2"])
    from PyPDF2 import PdfReader

def extract_text(filename, output_name):
    print(f"Extracting {filename}...")
    reader = PdfReader(filename)
    text = ""
    for i, page in enumerate(reader.pages):
        text += f"\n--- Page {i+1} ---\n"
        text += page.extract_text()
    
    with open(output_name, "w", encoding="utf-8") as f:
        f.write(text)
    print(f"Extracted {len(reader.pages)} pages to {output_name}")

extract_text("CV-park.pdf", "cv_park_text.txt")
extract_text("CV-choi.pdf", "cv_choi_text.txt")

File: scratch/extract_cvs.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function extractText(filename) {
    const dataBuffer = fs.readFileSync(filename);
    try {
        let parser = typeof pdf === 'function' ? pdf : (typeof pdf.default === 'function' ? pdf.default : null);
        if (!parser && pdf.PDFParse) parser = pdf.PDFParse;
        
        const data = await parser(dataBuffer);
        return data.text;
    } catch (error) {
        console.error(`Error parsing ${filename}:`, error);
        return null;
    }
}

async function main() {
    const parkText = await extractText('CV-park.pdf');
    const choiText = await extractText('CV-choi.pdf');

    if (parkText) fs.writeFileSync('cv_park_text.txt', parkText);
    if (choiText) fs.writeFileSync('cv_choi_text.txt', choiText);

    console.log('Extracted text to cv_park_text.txt and cv_choi_text.txt');
}

main();
